src/tyr/actor.cc

```diff
@@ -18,6 +18,10 @@ struct actor_t::pimpl_t {
       : reader(new baldr::GraphReader(config.get_child("mjolnir"))), loki_worker(config, reader),
         thor_worker(config, reader), odin_worker(config) {
   }
+  pimpl_t(const boost::property_tree::ptree& config, baldr::GraphReader& graph_reader)
```

Comment:
this other constructor allows passing a graph reader to actor instead of having it own the reader